File: find-the-duplicate-number.java
class Solution {
    public int findDuplicate(int[] nums) {
        
        HashSet<Integer> duplicateSet = new HashSet<>();

        for (int i = 0; i < nums.length; i++) {
            if (duplicateSet.contains(nums[i])) {
                return nums[i];
            }
            duplicateSet.add(nums[i]);
        }
        
        return 0;
    }
}

File: find-the-duplicate-number.py
class Solution(object):
    def findDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums_set = set()
        for n in nums:
            if n in nums_set:
                return n
            nums_set.add(n)
        return 0
